[10주차] 9조 FE 코드리뷰 7회차

seung365 opened the pull request:
## **PR**

### ✨ 작업 내용

- 마이페이지 UI 및 API 구현 되어 있는 부분 위주로 구현하였습니다. 
- 채팅 기능을 위해 sockjs-client 라이브러리를 사용했습니다. 웹소켓 연결되는 데까지 확인했습니다.
  - 메시지 수신이나 전송 등 다른 기능은 아직 구현 중입니다. 
- Discover 페이지도 구현 중입니다. UI는 만들어두었고, API 연동 중이었는데, 이 과정에서 문제가 ...
- s3를 통해 배포 후 ci를 통해 Master에 push 되면 자동으로 배포되게 하였습니다. (cloud front도 연결해놓았는데 아직 백엔드에서 http를 쓰고있어 description에는 http 링크만 걸려있습니다

### 🧐 질문

- Discover에서 다음과 같은 에러가 납니다. 
<img width="668" alt="image" src="https://github.com/user-attachments/assets/773ff1ce-85c3-4e6a-8d93-6d84d0690fd0">

data가 pages 배열 형식이라 해서 이렇게 참조하려고 하면 안 되네요...
솔직히 gpt 참고 많이 한 코드긴 한데.. 어떻게 수정해야 할지 모르겠습니다...😢
타입이 어디가 깨진 건지...ㅜㅠ

joojjang committed:
chore(Mygallery): 폴더 삭제

joojjang committed:
chore(MyGallery): 새 폴더 및 index 파일 생성 (대소문자 변경)

joojjang committed:
feat(Chat): chatList를 스테이트로 관리

joojjang committed:
Merge branch 'Weekly8' into feat/#71-chat-api

joojjang committed:
fix(Chat): 훅은 함수 컴포넌트 안에서 호출

joojjang committed:
Merge branch 'Review' into feat/#71-chat-api

joojjang committed:
feat(ChatRoom): 챗룸 페이지 만들고 라우팅, 채팅 목록 컴포넌트는 ChatList로 이름 변경

joojjang committed:
fix(routes): chat -> chats 경로명 변경

joojjang committed:
fix(routes): chat -> chats 경로명 변경

joojjang committed:
feat(chats): 챗리스트에서 챗룸 라우팅

joojjang committed:
fix(routes): 챗룸은 FreeLayout 자식으로

joojjang committed:
feat(ChatRoom): 페이지에 헤더 추가

- 헤더에 들어갈 아이콘으로 IconButton에 케밥 아이콘 추가

joojjang committed:
feat(ChatInput): 채팅 인풋창 구현

joojjang committed:
fix(reset.css): 적용 안 되는 속성 고침

- background-color: none 적용 안 됨

joojjang committed:
design(ChatInput): CSS 수정

joojjang committed:
feat(reset.css): textarea 기본 CSS 설정

joojjang committed:
feat(ChatInput): 자동 높이 조정

- useRef를 사용하여 textarea의 변화 감지
- 콜백 함수로 컴포넌트의 높이 chatInputHeight을 부모 컴포넌트로 전달
- 근데 높이 다시 지정해야 할 것 같음 안 맞음

joojjang committed:
feat(SearchBar): 서치 바 포지션 fixed로 변경, z-index 설정

joojjang committed:
feat(Discover): 이미지 목데이터

joojjang committed:
feat(SearchBar): 백그라운드 컬러 설정

joojjang committed:
feat(useGetFeed): 둘러보기 API 구현

joojjang committed:
refactor(SearchBar): sort imports

joojjang committed:
style(SearchBar): 공백 라인 추가

joojjang committed:
feat(ChatInput): message가 공백만으로 이루어져 있을 때 버튼 disabled

joojjang committed:
feat(MessageItem): 메시지 버블 및 날짜 표시 UI 구현 / ProfileImage 모듈화

joojjang committed:
design(MessageItem): CSS 수정

seung365 committed:
feat(api): useDeleteFollow 구현

seung365 committed:
feat(api): 팔로잉 목록 조회 api 구현

seung365 committed:
feat(api): 유저 일반 정보 조회 api 구현

seung365 committed:
feat(api): 아티스트 팔로우 api 구현

joojjang committed:
feat(chats API): index 파일 생성

joojjang committed:
Merge pull request #84 from kakao-tech-campus-2nd-step3/feat/#81-chatroom

채팅룸 페이지 구현 (API 빼고) #81

joojjang committed:
refactor(utils): 문자열 공백 제거 함수, 공백 제외 문자열 길이 세는 함수 생성

joojjang committed:
style(utils): formatDate 함수 형태 수정 (화살표 안 쓰고 function 키워드 사용)

seung365 committed:
feat(api): APIResponse 수정

seung365 committed:
refactor(CategoryTabBar): categorytabbar 공통 컴포넌트 수정

seung365 committed:
feat(type): api response type 추가

seung365 committed:
feat(SwiperFrame): 팔로우 기능 추가

seung365 committed:
feat(api): useGetFollow refetch 추가

seung365 committed:
feat(component): ArtistItem에 isFollow 여부 추가

seung365 committed:
feat(page): My page 구현

seung365 committed:
feat(component): My page 의 ProfileBox 구현

seung365 committed:
feat(component): My page의 MenuSection 구현

seung365 committed:
chore(component): FABContainer mode artist로 수정

seung365 committed:
feat(page): MyFavorite 페이지 구현

seung365 committed:
feat(component): swiperframe에 isFollow 여부 추가

seung365 committed:
feat(page): SearchResults 페이지에 category list 추가

seung365 committed:
feat(component): ArtistContents에 팔로우 api 연결

seung365 committed:
build: s3 배포 ci 추가

seung365 committed:
chore(gitignore): dist 추가

seung365 committed:
Merge branch 'Master' into Weekly10

joojjang committed:
feat(chats): 웹소켓 연결, 클라이언트 생성, sub/pub 구현

- SockJS 사용하여 웹소켓 연결 설정
- StompJS 설치하고 STOMP 클라이언트 생성
- 채팅 페이지 컴포넌트에서 메시지 구독, 메시지 전송

seung365 committed:
feat(api): useGetUser user일때와 artist에 따라 api 호출 다르게

seung365 committed:
feat(page): my page mode에 따라 다르게 렌더링

seung365 committed:
feat(component): profilebox 대신 user와 artist에 따른 profilebox 생성

seung365 committed:
feat(type): artist일때 response type 추가

seung365 committed:
feaT(component): MenuSection에서 User와 Artist일 경우 따로 분리

seung365 committed:
design(page): 팔로우 한 작가 없을 경우 UI 처리

seung365 committed:
feat(api): useGetUser type 수정

joojjang committed:
fix(vite.config.js): sockjs-client가 브라우저에서 구동되도록 수정

- sockjs-client가 global 변수를 찾을 수 있도록 window를 global로 설정해줌

joojjang committed:
chore(vite.config.js): 주석 추가

joojjang committed:
style(chats API): 함수에 function 키워드 사용

joojjang committed:
remove(ChatRoom): 임시 코드 삭제

joojjang committed:
chore(chats API): 주석 수정

joojjang committed:
refactor(chats): 메시지 타입 정의

seung365 committed:
chore(type): user interface 대신 type으로 수정

seung365 committed:
chore(api): useGetFollow export 수정

seung365 committed:
refactor(api): useGetUser type 선언되어있는거 재사용

seung365 committed:
design(component): 컴포넌트들 css 수정

seung365 committed:
feat(page): my page wrapper 추가

seung365 committed:
chore(api): useGetUser export 수정

Co-authored-by: Kim Minjoo <itchu999@naver.com>

joojjang committed:
Merge pull request #86 from kakao-tech-campus-2nd-step3/feat/#77

마이 페이지 UI 구현 및 API 연결 #77

joojjang committed:
Merge pull request #89 from kakao-tech-campus-2nd-step3/feat/#71-chat-api

채팅 - UI 구현 및 WebSocket 연결 #71 #81

joojjang committed:
fix(useGetFeed): API 연동하려고 시도 ...

joojjang committed:
fix(useGetFeed): 응답 타입 다시 수정

joojjang committed:
Merge branch 'Weekly10' into feat/#82-discover

joojjang committed:
Merge pull request #92 from kakao-tech-campus-2nd-step3/feat/#82-discover

둘러보기(discover) 페이지 구현 #82 #90

yujo11 commented on src/apis/products/useGetFeed.ts:
```diff
@@ -0,0 +1,51 @@
+  } catch (error) {
+    if (isAxiosError(error)) {
+      if (error.response) {
+        throw new Error(error.response.data.message || '피드 가져오기 실패');
+      } else {
+        throw new Error('네트워크 오류 또는 서버에 연결할 수 없습니다.');
+      }
+    } else {
+      throw new Error('알 수 없는 오류입니다.');
```
여기에서 throw 한 error는 어디에서 catch 하나요?

yujo11 commented on src/apis/users/useDeleteFollow.ts:
```diff
@@ -0,0 +1,28 @@
+  const response = await fetchInstance().delete(`/users/following/${artistId}`, {
+    headers: {
+      Authorization: `Bearer ${token}`,
+    },
```
이 부분이 반복되는 걸로 보이는데 반복되는 코드를 줄여볼 수 있을까요?

yujo11 commented on src/apis/users/useGetFollow.ts:
어디에서는 fetch  로직을 가지고 있는데서 error 처리를 하고, 어디에서는 react-query를 사용한 hook에서 에러 처리를 하고 있고, 여기에서는 error 처리를 하지 않고 있는데요. 

전체적으로 코드의 일관성을 지키면서 에러 핸들링 하는 방법에 대해 고민해보시면 좋을거 같아요

joojjang replied:
둘이서 나눠서 하다보니 서로 스타일이 달랐는데, 리팩토링 해보겠습니다 :)

yujo11 commented on src/apis/users/useGetFollow.ts:
```diff
@@ -0,0 +1,23 @@
+
+export const useGetFollow = () => {
+  const { data, status, refetch } = useQuery<APIResponse<FollowResponse>, Error>({
+    queryKey: ['followList'],
```
이런 queryKey도 상수화 해두면 나중에 invalidQueries 같은 로직을 통해 data를 refetch 해올 때 용이할거 같네요

yujo11 commented on src/apis/users/useGetUser.ts:
```diff
@@ -0,0 +1,26 @@
+
+type InfoType<T extends Mode> = T extends 'user' ? UserInfo : ArtistInfo;
+
+const getUser = async <T extends Mode>(mode: T): Promise<APIResponse<InfoType<T>>> => {
+  const endpoint = mode === 'user' ? '/users' : '/artists';
+  const response = await fetchInstance().get(endpoint, {
+    headers: {
+      Authorization: `Bearer ${token}`,
+    },
+  });
+  return response.data;
+};
+
+export const useGetUser = <T extends Mode>(mode: T) => {
+  const { data, isLoading, isError } = useQuery<APIResponse<InfoType<T>>, Error>({
+    queryKey: ['userInfo', mode],
+    queryFn: () => getUser(mode),
+  });
+
+  return { data, isLoading, isError };
+};
```
getUsers를 하는 함수와 getArtists를 하는 함수를 분리해서 선언하고 사용하는 곳에서 상황에 맞게 사용할 수 있도록 만들어보셔도 좋을거 같네요

seung365 replied:
아 넵 이부분 저도 처음에는 분리했는데 User 정보가 둘 중 하나라 하나로 처리해보긴 했습니다.
좀 어색하다고 느끼신거 같아 다시 분리하겠습니다!

yujo11 commented on src/components/common/CategoryTabBar/index.tsx:
```diff
@@ -1,35 +1,63 @@
+      {tabList.map((category, index) => (
+        <TabWrapper
+          key={index}
+          isActive={onActive === category}
```
tabState 그대로 `onActive` state에 들어가고 있는거 같은데 아래처럼 작성해보는건 어떨까요?

```suggestion
          isActive={tabState === category}
```

yujo11 commented on src/pages/Discover/index.tsx:
```diff
@@ -1,5 +1,76 @@
+    <Wrapper>
+      <SearchBar />
+      <ContentWrapper>
+        <Suspense fallback={<>Loading...</>}>
```
👍 👍 👍

joojjang committed:
chore(index.html): 파비콘 변경, 타이틀 변경

yujo11 commented on src/pages/My/components/UserProfileBox.tsx:
```diff
@@ -0,0 +1,91 @@
+      <DetailWrapper>
+        <NameWrapper>{username}</NameWrapper>
+        <TagWrapper>
+          {' '}
```
여긴 의도적으로 공백을 넣으신 부분인가요? 코드의 의도 전달이 잘 되지 않아 공백문자를 대신해서 사용하셔도 좋을거 같네요

yujo11 commented on src/pages/My/index.tsx:
```diff
@@ -1,12 +1,63 @@
+  const { mode } = useModeStore();
+  const { data, isLoading, isError } = useGetUser(mode);
+
+  if (isLoading) {
+    return <p>Loading...</p>;
+  }
```
일관되게 Suspense를 이용해 로딩 처리가 되도록 구현해보셔도 좋을거 같네용

joojjang committed:
feat(z-index): 필요한 z-index 값 추가 탭바, fab)

joojjang committed:
design(ChatItem): 마우스 hover와 active에 따라 UI 변화

yujo11 reviewed (changes requested):
안녕하세요 이번 주도 고생 많으셨습니다 ~

에러가 발생할 때는 에러 메세지에 주목해서 하나씩 따라가보면 좋은데요

<img width="549" alt="image" src="https://github.com/user-attachments/assets/d2685281-aeae-4f0e-ac2c-70156324dbe2">

현재 `data.pages`가 타입이 추론되지 않고 있고 useGetFeed를 확인해보면 아래와 같이 data가 void 형식으로 추론되는 걸 확인할 수 있는데요

<img width="651" alt="image" src="https://github.com/user-attachments/assets/eebe2efb-fd7a-4aad-9601-3735141edf57">

따라서 useGetFeed에 들어가보면 여기 역시 반환 타입 추론이 제대로 되지 않는걸 확인할 수 있습니다 

<img width="681" alt="image" src="https://github.com/user-attachments/assets/784ddf1b-76c0-45d9-a94f-4a9c1a503a16">


useGetFeed의 반환타입이 원하는대로 추론되도록 이 부분을 수정해보시면 좋을거 같네요

joojjang committed:
fix(useGetFeed): useSuspenseInfiniteQuery의 타입 명시하지 않음

- useSuspenseInfiniteQuery는 isPlaceholderData 속성을 포함하지 않아 UseInfiniteQueryResult 타입과 호환되지 않는 오류가 발생
- 알아서 타입 추론 되도록 둠

joojjang committed:
fix(useGetFeed): 요청 경로에서 page 파라미터 삭제

joojjang committed:
Merge pull request #96 from kakao-tech-campus-2nd-step3/fix/#95-weekly10-review

Weekly10 에러 픽스 #95

joojjang committed:
refactor(API instance): 모든 요청에 token 포함하여 보내기

joojjang committed:
fix(API instance): timeout 시간 짧게 (3초)

joojjang committed:
refactor(API instance): API 요청 시 최신 토큰 유지

joojjang committed:
refactor(apis): fetchInstance와 queryClient 파일 분리

joojjang committed:
style(apis): 코드 스타일 맞춤

- default 컴포넌트
- 함수 선언식 사용
- import문 순서 정리

joojjang committed:
fix: import문 문법 에러 픽스

joojjang committed:
fix(instance): 외부 API 요청에 Authorization 헤더 미포함

joojjang committed:
feat(Discover): ErrorBoundary 사용해서 에러 폴백 처리

joojjang committed:
feat(queryKeys): 쿼리 키 모아서 상수화

seung365 committed:
chore(package): react-error-boundary 설치

joojjang committed:
refactor(constants/zIndex): zIndex 파일 위치 이동

seung365 committed:
refactor(api): User과 Artist로 api 분리

seung365 committed:
refactor(page): My page suspense와 errorboudary로 리팩토링

joojjang committed:
refactor(fetchInstance): 파일명 변경

joojjang committed:
refactor(styles/zIndex): zIndex 파일 위치 다시 이동

seung365 committed:
chore(UserProfileBox): 빈 공백 삭제

seung365 committed:
refactor(api): queryKey 상수화

joojjang committed:
Merge pull request #100 from kakao-tech-campus-2nd-step3/refactor/#97-weekly10

weekly10 리팩토링 #97

seung365 committed:
chore: 충돌 해결

joojjang commented:
질문 드렸던 타입 오류 덕분에 고쳤습니다!

그리고
getFeed 함수에서 throw하는 에러는 useGetFeed에서 서스펜스가 캐치하고, 이를 ErrorBoundary로 전달해서 폴백으로 처리하는 로직을 구상했는데  
<img width="431" alt="image" src="https://github.com/user-attachments/assets/b3029885-11bd-4aa6-a191-ce8bc6cae926">
<img width="763" alt="image" src="https://github.com/user-attachments/assets/ad9c4aa3-4f66-4064-9d45-b80504858289">

화면이랑 콘솔이 이렇게 뜨는 걸 보면.. 에러 바운더리로 잘 처리된 게 맞는 걸까요?..  
gpt한테 물어보니까 
> ErrorBoundary는 전체 컴포넌트 트리에 영향을 미쳐서, 에러가 발생하면 해당 범위 내의 모든 UI를 fallback UI로 교체합니다.  
즉, Suspense 내부에서 에러가 발생하면 ErrorBoundary가 fallback UI로 전체 화면을 덮어버릴 수 있습니다.

즉 에러가 처리되기 전까지 전체 화면에 아무 것도 뜨지 않고 폴백 UI만 뜬다... 로 이해하는 게 맞을까요..?

정리하자면 궁금한 게 두 가지입니다.  
1. getFeed에서의 에러가 Discover에서 에러바운더리로 처리가 잘 된 게 맞는 건지,
2. 에러가 나면 에러바운더리로 감싸진 부분만 폴백 ui로 대체되는 게 아니라 전체 화면이 대체되는 게 맞는 건지 (저는 에러가 나더라도 에러바운더리 밖에 있는 SearchBar는 보여야되는 거 아닌가 싶은데..)  

근데 생각해보니 폴백으로 준 'Error' 글자도 안 떠서 그냥 제대로 처리 안 된 것 같네요...

seung365 committed:
refactor(api): fetchInstance 수정으로 인한 api 코드 수정

seung365 committed:
Merge pull request #101 from kakao-tech-campus-2nd-step3/refactor/#99-feedback

10주차 코드 리뷰 피드백 #99

seung365 committed:
refactor(api): api 요청 함수 형식 통일

seung365 committed:
Merge pull request #102 from kakao-tech-campus-2nd-step3/refactor/#99-feedback

추가 feedback 반영 #99

joojjang committed:
Merge branch 'Weekly10'

joojjang committed:
refactor(constants): 폴더 구조 리팩토링, heights 파일 생성

joojjang committed:
refactor(styles/constants): CSS 설정에 필요한 상수들은 styles 디렉토리로 이동

joojjang committed:
Merge pull request #104 from kakao-tech-campus-2nd-step3/refactor/#103-constants

constants, styles 폴더 정리 #103

seung365 committed:
chore(api): useGetArtist api 폴더 위치 변경

seung365 committed:
fix(MyPage): my page suspense, errorBoundary 에러 수정

seung365 commented:
> 질문 드렸던 타입 오류 덕분에 고쳤습니다!
> 
> 그리고 getFeed 함수에서 throw하는 에러는 useGetFeed에서 서스펜스가 캐치하고, 이를 ErrorBoundary로 전달해서 폴백으로 처리하는 로직을 구상했는데 <img alt="image" width="431" src="https://private-user-images.githubusercontent.com/91789311/384668725-b3029885-11bd-4aa6-a191-ce8bc6cae926.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEyMzMwMjMsIm5iZiI6MTczMTIzMjcyMywicGF0aCI6Ii85MTc4OTMxMS8zODQ2Njg3MjUtYjMwMjk4ODUtMTFiZC00YWE2LWExOTEtY2U4YmM2Y2FlOTI2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTEwVDA5NTg0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUzYTUyM2QyMmZjOWEzNTEyY2MxOWI4MzVjYzVkNWEwYzU3MDRkYzMyYmQyOTQwNGFlMGNhM2MwZmNjMDNiMTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.YRIL3eI-7JxeC3DLiDyO6IXFXjlUCcBvhnUexXkTFq4"> <img alt="image" width="763" src="https://private-user-images.githubusercontent.com/91789311/384669301-ad9c4aa3-4f66-4064-9d45-b80504858289.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEyMzMwMjMsIm5iZiI6MTczMTIzMjcyMywicGF0aCI6Ii85MTc4OTMxMS8zODQ2NjkzMDEtYWQ5YzRhYTMtNGY2Ni00MDY0LTlkNDUtYjgwNTA0ODU4Mjg5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTEwVDA5NTg0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWExMjJhMWViZTRkMWVmZmIzOTQ3NDkxYjE5ZTQ3YWU1MGI5NWUwOGQyYTg0ZDZjZjI2ZmVlNDQzMzEyZjhjNmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.LgLx2ic8LFf2HpTapO_GLR2v06ZlnPq-mykYHVA0Htk">
> 
> 화면이랑 콘솔이 이렇게 뜨는 걸 보면.. 에러 바운더리로 잘 처리된 게 맞는 걸까요?.. gpt한테 물어보니까
> 
> > ErrorBoundary는 전체 컴포넌트 트리에 영향을 미쳐서, 에러가 발생하면 해당 범위 내의 모든 UI를 fallback UI로 교체합니다.
> > 즉, Suspense 내부에서 에러가 발생하면 ErrorBoundary가 fallback UI로 전체 화면을 덮어버릴 수 있습니다.
> 
> 즉 에러가 처리되기 전까지 전체 화면에 아무 것도 뜨지 않고 폴백 UI만 뜬다... 로 이해하는 게 맞을까요..?
> 
> 정리하자면 궁금한 게 두 가지입니다.
> 
> 1. getFeed에서의 에러가 Discover에서 에러바운더리로 처리가 잘 된 게 맞는 건지,
> 2. 에러가 나면 에러바운더리로 감싸진 부분만 폴백 ui로 대체되는 게 아니라 전체 화면이 대체되는 게 맞는 건지 (저는 에러가 나더라도 에러바운더리 밖에 있는 SearchBar는 보여야되는 거 아닌가 싶은데..)
> 
> 근데 생각해보니 폴백으로 준 'Error' 글자도 안 떠서 그냥 제대로 처리 안 된 것 같네요...

<img width="695" alt="image" src="https://github.com/user-attachments/assets/94b541c5-ef31-40f0-bbf7-246c2f48366d">
<img width="455" alt="image" src="https://github.com/user-attachments/assets/d6a15e24-f89f-4726-a3d1-f6c2280c95c4">

제가 이해한 바로는 suspense와 errorboundary가 데이터 fetching이 시작되기전에 준비가 되어있어야 한다고 이해해서
이런식으로 밖에서 감싸면 에러 해결 될꺼 같아 남겨요!

joojjang committed:
fix(Discover): ErrorBoundary 제대로 안 됐던 오류 수정

- data 페칭 전에 ErrorBoundary와 Suspense가 준비되도록

joojjang committed:
Merge pull request #106 from kakao-tech-campus-2nd-step3/fix/#105-errorboundary

Discover 에러바운더리 오류 픽스 #105

joojjang committed:
Merge branch 'feat/#93-chat-api' into Weekly10

seung365 committed:
chore(api): useGetArtist 위치 이동

yujo11 commented on src/apis/artists/useGetArtist.ts:
```diff
@@ -0,0 +1,20 @@
+  return response.data;
+}
+
+const useGetArtist = () => {
+  const { data } = useSuspenseQuery<APIResponse<ArtistInfo>, Error>({
+    queryKey: [QueryKeys.USER_INFO],
+    queryFn: getArtist,
+  });
```
`options`를 사용하는 곳에서 넣을 수 있게 하면 더 유연하게 사용이 가능할거 같네요

yujo11 replied:
react-query v5 에서는 queryOptions 같은걸 사용해서 option만 추상화해서 사용하는 방법도 많이들 사용하는거 같아요 참고 하시면 좋겠네요 ㅎㅎ

yujo11 reviewed (approved):
고생 많으셨습니다~ 소소한 코멘트 하나 남겼는데 확인해보시면 좋을거 같아요 ㅎㅎ
